wgpu/src/quad.rs

```diff
@@ -167,119 +89,128 @@ impl Pipeline {
         device: &wgpu::Device,
         staging_belt: &mut wgpu::util::StagingBelt,
         encoder: &mut wgpu::CommandEncoder,
-        instances: &[layer::Quad],
+        instances: &layer::Quads,
         transformation: Transformation,
         scale: f32,
         bounds: Rectangle<u32>,
         target: &wgpu::TextureView,
     ) {
-        let uniforms = Uniforms::new(transformation, scale);
+        // write uniforms to GPU
+        let _ = self.uniforms.write(
+            device,
+            staging_belt,
+            encoder,
+            0,
+            &[Uniforms::new(transformation, scale)],
+        );
+
+        // resize buffers if necessary
+        let _ = self.solid.instances.resize(device, instances.solids.len());
+        let _ = self
+            .gradient
+            .instances
+            .resize(device, instances.gradients.len());
```

Comment:
So we are resizing the buffers now instead of looping and uploading multiple times.

I think that should be fine, and it fits the architecture of the incoming overhaul. Maybe we can reduce the `MAX_INSTANCES` to `10_000` here as well.

Reply:
We should probably rename `MAX_INSTANCES` to `INITIAL_INSTANCES` or similar.